crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -40,10 +40,14 @@ impl<'interner> TypeChecker<'interner> {
 
                         let first_elem_type = elem_types.get(0).cloned().unwrap_or(Type::Error);
 
-                        let arr_type = Type::Array(
-                            Box::new(Type::Constant(arr.len() as u64)),
-                            Box::new(first_elem_type.clone()),
-                        );
+                        let arr_type = if arr.is_empty() && self.interner.enable_slices {
+                            Type::Slice(Box::new(self.interner.next_type_variable()))
```

Comment:
Why do we infer a slice type if the array is empty rather than an array of length zero?

Reply:
I switched to an empty array. Previously we didn't allow arrays of zero length at all so I was making zero-length arrays slices but that is unnecessary. Lmk what you think of the switch